### PR TITLE
Count_Inversion.java

### LINKED ISSUE #1664: Count inversion in c++ and java.
The code in the repo are given in c and js. Please assgin me this for c++ and java.

### PR DESCRIPTION
**Pull request by mansikagrawal**
Fixes #1664 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Files Added: Count_Inversion.java

Thanks!

### CODE REVIEW

File: Count_Inversion/Count_Inversion.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	
+	public static void main(String[] args) 
+	{ 
+	    
+	    int n;
+	    Scanner in=new Scanner(System.in);
+	    n=in.nextInt();
+	    int a[]=new int[n];
+	    for(int i=0;i<n;i++)
+	    a[i]=in.nextInt();
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please add space around the operator.

File: Count_Inversion/Count_Inversion.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+	
+	public static void main(String[] args) 
+	{ 
+	    
+	    int n;
+	    Scanner in=new Scanner(System.in);
+	    n=in.nextInt();
+	    int a[]=new int[n];
+	    for(int i=0;i<n;i++)
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please add space around the operator

File: Count_Inversion/Count_Inversion.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		return invcount; 
+	} 
+
+	
+	public static void main(String[] args) 
+	{ 
+	    
+	    int n;
+	    Scanner in=new Scanner(System.in);
+	    n=in.nextInt();
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please add space around the operator

File: Count_Inversion/Count_Inversion.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+		return invcount; 
+	} 
+
+	
+	public static void main(String[] args) 
+	{ 
+	    
+	    int n;
+	    Scanner in=new Scanner(System.in);
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please add space around the operator

File: Count_Inversion/Count_Inversion.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+	} 
+
+	/* This method merges two sorted arrays and returns inversion count in 
+	the arrays.*/
+	static int mergefn( int arr[], int temp[], int left, int mid, int right ) 
+	{ 
+		int i, j, k; 
+		int invcount = 0; 
+
+		i = left; /* i is index for left subarray*/
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Add comment above the statement

File: Count_Inversion/Count_Inversion.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+	/* This function returns  the number of inversions in the array */
+	static int mergeSort( int arr[], int temp[], int left, int right ) 
+	{ 
+		int mid, invcount = 0; 
+		if (right > left) { 
+			/* Divide the array into two parts */
+			mid = (right + left) / 2; 
+
+		    /* inversions in the left subarray */
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please correct th comment indentation

File: Count_Inversion/Count_Inversion.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+		System.out.println(sortfn(a, n)); 
+	} 
+} 
+
+
+
+/* Sample input
+5 
+1 20 6 4 5
+Sample output
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Leave a space before output

**Comment by mansikagrawal** [comment]
@somya-kapoor  Mam please check the issue again I had done the changes.

File: Count_Inversion/Count_Inversion.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+	
+	public static void main(String[] args) 
+	{ 
+	    
+	    int n;
+	    Scanner in = new Scanner(System.in);
+	    n = in.nextInt();
+	    int a[] = new int[n];
+	    for(int i=0; i<n; i++)
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please add space around the operator.

File: Count_Inversion/Count_Inversion.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+	    a[i] = in.nextInt();
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please correct the indentation.

File: Count_Inversion/Count_Inversion.java [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+		System.out.println(sortfn(a, n));
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please correct the indentation.

**Comment by somya-kapoor** [comment]
Please use 4 space indentation.

**Comment by mansikagrawal** [comment]
@somya-kapoor  Mam please check any further changes required.

**Comment by somya-kapoor** [comment]
@mansikagrawal Please use 4 space indentation.

**Comment by mansikagrawal** [comment]
@somya-kapoor Mam please check the last commit.  Any changes required?

**Comment by somya-kapoor** [comment]
@mansikagrawal Please use 4 space indentation, you are using 8 space indentation.

**Pull request closed without merging**